ass4.cpp

```diff
@@ -0,0 +1,73 @@
+#include <iostream>
+#include <vector>
+#include <deque>
+
+using namespace std;
+
+// First variant of algorithm using recursion
+
+void flood_fill_recursive(vector<vector<bool> > &table, int y, int x) {
```

Comment:
A pointer is better as a parameter you want to modify, make the calling code more explicit, compare:
`flood_fill_recursive(table, y, x)
`vs
`flood_fill_recursive(&table, y, x)`
In the second is obvious that the table will be modified.